pkg/tuple/tuple.go

```diff
@@ -201,15 +207,25 @@ func Parse(tpl string) *core.RelationTuple {
 		}
 	}
 
+	resourceID := groups[stringz.SliceIndex(parserRegex.SubexpNames(), "resourceID")]
+	if err := ValidateResourceID(resourceID); err != nil {
+		return nil
+	}
+
+	subjectID := groups[stringz.SliceIndex(parserRegex.SubexpNames(), "subjectID")]
+	if err := ValidateSubjectID(subjectID); err != nil {
+		return nil
+	}
+
```

Comment:
I believe this is a redundant since `core.proto` already generates validation logic in `pkg/proto/core/v1/core.pb.validate.go`. We would be keeping 2 copies of regex that could drift-away, and do redundant and expensive regex checks.

Reply:
This hasn't changed and I feel it is important to have this extra check in case the validation code fails for some reason, especially since the validation code could be removed externally without "SpiceDB" knowing

Reply:
it's true but we get the same level of confidence by testing all these things in SpiceDB. Removing this does not make this less secure IMO, and likewise I think the code is not more secure by having this here.

Reply:
We actually don't: the constructed relationship/tuple is not having its `.Validate` called here. Even if we did, we'd need to keep `ValidateSubjectID` and the other methods around *anyway* because the `.Validate` method returns poor error messages, so we have to validate in a better fashion regardless

Reply:
it sounds like the only problem here is the message returned by the generate `Validate` method?